hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
Right. But the problem is memcpy (or what the compiler inlines in place of memcpy) could be doing any access size. It will probably(TM) do the access sizes you expect if given nice aligned sizes, but there's no guarantee.

In this sort of application, it's normal to have custom transfer routines that enforce the access size. Which also means "volatile" (otherwise it could conceivably unroll the code and use 64-bit accesses).

On the same subject, what about the flash cache? Does that require care during programming? Maybe @c1728p9 can advise if there's special handling required.